indra/llcommon/llerror.h

```diff
@@ -308,15 +308,23 @@ namespace LLError
     class LLUserWarningMsg
     {
     public:
-        typedef std::function<void(const std::string&, const std::string&)> Handler;
+        typedef enum
+        {
+            ERROR_OTHER = 0,
+            ERROR_BAD_ALLOC = 1,
+            ERROR_MISSING_FILES = 2,
+        } eLastExecEvent;
+
+        // tittle, message and error code to include in error marker file
+        typedef std::function<void(const std::string&, const std::string&, S32 error_code)> Handler;
         static void setHandler(const Handler&);
         static void setOutOfMemoryStrings(const std::string& title, const std::string& message);
 
         // When viewer encounters bad alloc or can't access files try warning user about reasons
         static void showOutOfMemory();
         static void showMissingFiles();
         // Genering error
-        static void show(const std::string&);
+        static void show(const std::string&, S32 error_code = -1);
```

Comment:
If `ERROR_OTHER` is the "something else went wrong" code, wouldn't we want `ERROR_OTHER` to be the default value for `show(error_code)`?

Reply:
At the moment -1 means 'don't create a marker' in case you only need only a message. The idea was that LL_ERRS will be creating llerror marker or more likely writing own ERROR_LLERROR flag into the marker (but time will tell how it turns out).

Reply:
May be I should have created a separate createMarkerFile in LLError instead of reusing show().